webpack.config.js

```diff
@@ -1,18 +1,113 @@
 const path = require("path");
+const copy = require("copy-webpack-plugin");
+const fs = require("fs");
 
-module.exports = {
-    entry: ["./src/index.js"],
-    output: {
-        filename: "bundle.js",
-        path: path.resolve(__dirname, "build")
+/* These can be overridden, typically from the Makefile.am */
+const srcdir = (process.env.SRCDIR || __dirname) + path.sep + "src";
+const builddir = (process.env.SRCDIR || __dirname);
+const distdir = builddir + path.sep + "dist";
+const section = process.env.ONLYDIR || null;
+
+var info = {
+    entries: {
+        "index": [
+            "./index.js"
+        ]
     },
+    files: [
+        "index.html"
+    ]
+};
+
+var output = {
+    path: distdir,
+    filename: "[name].js",
+    sourceMapFilename: "[file].map",
+};
+
+/*
+ * Note that we're avoiding the use of path.join as webpack and nodejs
+ * want relative paths that start with ./ explicitly.
+ *
+ * In addition we mimic the VPATH style functionality of GNU Makefile
+ * where we first check builddir, and then srcdir.
+ */
+
+function vpath(/* ... */) {
+    var filename = Array.prototype.join.call(arguments, path.sep);
```

Comment:
I have not seen ES6 in use in webpack.config.js ... which is a Turing complete configuration file :S  My opinion: Tread cautiously with ES6 usage there.